automation/scripts/clean_pipelines/clean_pipelines.py

```diff
@@ -0,0 +1,392 @@
+
+        return formatted_date
+    except (ValueError, OverflowError) as e:
+        raise ValueError(
+            f"Invalid date format: {date_input}. Please provide a valid date."
+        ) from e
+
+
+def get_kfp_client(
+    kfp_url=mlrun.mlconf.kfp_url, namespace: str = "default-tenant"
+) -> Client:
+    kfp_client = mlrun_pipelines.utils.get_client(kfp_url, namespace)
+    return kfp_client
+
+
+def get_experiment_name(kfp_client: Client, experiment_id: str) -> str:
+    experiment = kfp_client.get_experiment(experiment_id=experiment_id)
+    return experiment.name if experiment else ""
+
+
+def filter_out_non_related_runs(
+    project_name: str, runs: list[PipelineRun]
+) -> list[PipelineRun]:
+    project_runs = []
+    for run in runs:
+        run_project = mlrun_pipelines.mixins.PipelineProviderMixin().resolve_project_from_workflow_manifest(
+            run.workflow_manifest()
+        )
+        if run_project == project_name:
+            project_runs.append(run)
+    return project_runs
+
+
+def get_list_runs_filter(project_name: str, end_date: str, start_date: str) -> str:
+    filters = {
+        "predicates": [
+            {
+                "key": "created_at",
+                "op": 7,  # Operation 7 corresponds to '<=' (less than or equal)
+                "timestamp_value": end_date,
+            },
+        ]
+    }
+    if not project_name == "*":
+        filters["predicates"].append(
+            {
+                "key": "name",
+                "op": 9,  # Operation 9 corresponds to substring matching
+                "string_value": project_name,
+            }
+        )
+    if start_date:
+        filters["predicates"].append(
+            {
+                "key": "created_at",
+                "op": 5,  # Operation 5 corresponds to '>=' (greater than or equal)
+                "timestamp_value": start_date,
+            }
+        )
+    return json.dumps(filters)
+
+
+def list_pipelines_runs(
+    kfp_client: Client, query_filter: str, page_token: str = "", sort_by: str = ""
+) -> list[PipelineRun]:
+    runs = []
+    counter = 0
+    while page_token is not None:
+        # kfp doesn't allow us to pass both a page_token and the `filter` and `sort_by` params.
+        # When we have a token from previous call, we will strip out the filter and use the token to continue
+        # (the token contains the details of the filter that was used to create it)
+        response = kfp_client.list_runs(
+            page_token=page_token,
+            page_size=mlrun.common.schemas.PipelinesPagination.max_page_size,
+            sort_by=sort_by if page_token == "" else "",
+            filter=query_filter if page_token == "" else "",
+        )
+        runs.extend([PipelineRun(run) for run in response.runs or []])
+        page_token = response.next_page_token
+        if counter % 50 == 0:
+            mlrun.utils.logger.info(
+                "Collecting pipelines runs:", runs_count=(10000 * counter)
```

Comment:
From where the `10000` comes?
Shouldn't it be `mlrun.common.schemas.PipelinesPagination.max_page_size * counter` ?

Reply:
Agree we need some explanation of these magic numbers (% 50 ?)